File: week2/p2.py
mat = [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8]
]
n = len(mat)

# Transpose the matrix
for i in range(n):
    for j in range(i + 1, n):
        mat[i][j], mat[j][i] = mat[j][i], mat[i][j]

# Reverse each column
for j in range(n):
    top, bottom = 0, n - 1
    while top < bottom:
        mat[top][j], mat[bottom][j] = mat[bottom][j], mat[top][j]
        top += 1
        bottom -= 1

# Print rotated matrix
for row in mat:
    print(row)

File: week2/p2.c
#include <stdio.h>

int main() {
    int n = 3;
    int mat[3][3] = {
        {0, 1, 2},
        {3, 4, 5},
        {6, 7, 8}
    };

    // Transpose the matrix
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int temp = mat[i][j];
            mat[i][j] = mat[j][i];
            mat[j][i] = temp;
        }
    }

    // Reverse each column
    for (int j = 0; j < n; j++) {
        int top = 0, bottom = n - 1;
        while (top < bottom) {
            int temp = mat[top][j];
            mat[top][j] = mat[bottom][j];
            mat[bottom][j] = temp;
            top++;
            bottom--;
        }
    }

    // Print rotated matrix
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            printf("%d ", mat[i][j]);
        }
        printf("\n");
    }

    return 0;
}
